src/Microsoft.NetCore.Analyzers/Core/Runtime/TestForNaNCorrectly.cs

```diff
@@ -53,7 +51,7 @@ public override void Initialize(AnalysisContext analysisContext)
                 operationAnalysisContext =>
                 {
                     var binaryOperatorExpression = (IBinaryOperatorExpression)operationAnalysisContext.Operation;
-                    if (!_comparisonOperators.Contains(binaryOperatorExpression.BinaryOperationKind))
+                    if (!_comparisonOperators.Contains(binaryOperatorExpression.OperatorKind))
                     {
```

Comment:
shouldn't be there code that checks type is float or not?

Reply:
It already used to check that down in the analyzer.